quizyapp/models.py

```diff
@@ -0,0 +1,100 @@
+from django.db import models
+from django.contrib.auth.models import User
+from hashlib import md5
+from html import unescape
+from django.db.models.fields.related import ForeignKey
+import requests
+
+class UserPoints(models.Model):
+    user = models.ForeignKey(User, on_delete=models.PROTECT) 
+    points = models.IntegerField()
+
+class Category(models.Model):
+    id = models.IntegerField(unique=True, primary_key=True)
+    name = models.TextField(unique=True)
+
+    @staticmethod
+    def init_category_list_from_api_if_none_available():
+        if not Category.objects.all().count():
+            category_dict = Category.get_category_dict_from_opentdb_api()
+            for id, name in category_dict.items():
+                Category.objects.create(id=id, name=name)
+
+    @staticmethod
+    def get_category_dict_from_opentdb_api():
+        d = {}
+        r = requests.get('https://opentdb.com/api_category.php')
+        for c in r.json()['trivia_categories']:
+            d[c['id']]=c['name']
+        return d
+
+    def __str__(self):
+        return f'id: {self.id}, name: {self.name}'
+
+class Answer(models.Model):
+    id = models.CharField(max_length=32, unique=True, primary_key=True)
+    text = models.TextField()
+    is_correct = models.BooleanField()
+
+    def save(self, **kwargs):
+        self.id = md5(''.join([self.text, str(self.is_correct)]).encode('utf-8')).hexdigest()
+        super().save(**kwargs)
+
+    def __str__(self):
+        return f'id: {self.id}, text: {self.text}, is_correct: {self.is_correct}'
+
+class Question(models.Model):
+    id = models.CharField(max_length=32, unique=True, primary_key=True)
+    answers = models.ManyToManyField(Answer)
```

Comment:
Ja bym to raczej widział jako 1:n, a nie wiele do wielu.

Reply:
teź myślałem o tym, ale np są pytania z odpowiedziami True/False i chciałem ich używać ponownie w różnych pytaniach. chociaż później przy sprawdzaniu może to trochę utrudnia sprawę kiedy dana odpowiedź nie jest przypisana jednoznacznie do pytania. ale zdecydowalem się na taki model, ze względu na to że będzie mniej danych gromadzone

Reply:
relacja 1:n jest rzeczywiście bradziej sensowna w tym przypadku, zaimplementowane!